sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -110,7 +110,11 @@ private[csv] class UnivocityParser(
       }
 
     case _: BooleanType => (d: String) =>
-      nullSafeDatum(d, name, nullable, options)(_.toBoolean)
+      nullSafeDatum(d, name, nullable, options) { datum =>
+        if (datum.equalsIgnoreCase(options.trueValue)) true
```

Comment:
Just seems more obvious that `datum` is guaranteed to be not null given the name of `nullSafeDatum`. Probably should switch around the values in `CSVInferSchema.scala`, since there, `field` is null checked at the beginning of the parsing chain. Do you prefer one over the other?